Exploratory Data Analysis (EDA)

Project: Predicting Career Domain and Seniority from LinkedIn Profiles
Goal is to Understand the structure and characteristics of Active job positions


In [1]:
import json
from pathlib import Path
import pandas as pd


In [ ]:
from pathlib import Path

PROJECT_ROOT = Path.cwd().parents[1]       # aktuelles Working Directory  #CAPSTONE-Project-/
PROJECT_ROOT



WindowsPath('c:/Capstone-Project-')

In [10]:
DATA_PATH = PROJECT_ROOT / "files"
FILE = DATA_PATH / "linkedin-cvs-annotated.json"
FILE.exists()

True

In [11]:
with open(FILE, "r", encoding="utf-8") as f:
    cvs = json.load(f)

type(cvs), len(cvs)

(list, 609)

Data Type: List
Count of CVs: 609


In [12]:
cvs[0]

[{'organization': 'Depot4Design GmbH',
  'linkedin': 'https://www.linkedin.com/company/depot4design-gmbh',
  'position': 'Prokurist',
  'startDate': '2019-08',
  'endDate': None,
  'status': 'ACTIVE',
  'department': 'Other',
  'seniority': 'Management'},
 {'organization': 'Depot4Design GmbH',
  'linkedin': 'https://www.linkedin.com/company/depot4design-gmbh',
  'position': 'CFO',
  'startDate': '2019-07',
  'endDate': None,
  'status': 'ACTIVE',
  'department': 'Other',
  'seniority': 'Management'},
 {'organization': 'Depot4Design GmbH',
  'linkedin': 'https://www.linkedin.com/company/depot4design-gmbh',
  'position': 'Betriebswirtin',
  'startDate': '2019-07',
  'endDate': None,
  'status': 'ACTIVE',
  'department': 'Other',
  'seniority': 'Professional'},
 {'organization': 'Depot4Design GmbH',
  'linkedin': 'https://www.linkedin.com/company/depot4design-gmbh',
  'position': 'Prokuristin',
  'startDate': '2019-07',
  'endDate': None,
  'status': 'ACTIVE',
  'department': 'Other',
  '

In [15]:
from collections import Counter

active_job_counts = []

for cv in cvs:
    count = sum(1 for job in cv if job.get("status") == "ACTIVE")
    active_job_counts.append(count)

Counter(active_job_counts)


Counter({1: 380, 0: 131, 2: 78, 3: 10, 5: 4, 4: 3, 8: 1, 7: 1, 10: 1})

Most individuals have 1 job, second most individuals have zero active jobs and so on...




In [20]:
import pandas as pd

jobs = []
for cv in cvs:
    for job in cv:
        jobs.append(job)

df = pd.DataFrame(jobs)
df.head()





,organization,linkedin,position,startDate,endDate,status,department,seniority
0,Depot4Design GmbH,https://www.linkedin.com/company/depot4design-...,Prokurist,2019-08,None,ACTIVE,Other,Management
1,Depot4Design GmbH,https://www.linkedin.com/company/depot4design-...,CFO,2019-07,None,ACTIVE,Other,Management
2,Depot4Design GmbH,https://www.linkedin.com/company/depot4design-...,Betriebswirtin,2019-07,None,ACTIVE,Other,Professional
3,Depot4Design GmbH,https://www.linkedin.com/company/depot4design-...,Prokuristin,2019-07,None,ACTIVE,Other,Management
4,Depot4Design GmbH,https://www.linkedin.com/company/depot4design-...,CFO,2019-07,None,ACTIVE,Other,Management


In [22]:
df_active = df[df["status"] == "ACTIVE"]
df_active.shape

(623, 8)

There are 623 jobs with the Status Active currently.

In [23]:
df[["department", "seniority"]].isna().mean()


department    0.0
seniority     0.0
dtype: float64

All job entries in the data set are fully labeled with both department and seniority information. No missing Values. 

In [24]:
df["department"].value_counts(normalize=True)


department
Other                     0.474602
Information Technology    0.118271
Sales                     0.083017
Consulting                0.073920
Project Management        0.066338
Marketing                 0.050417
Administrative            0.031842
Business Development      0.029568
Purchasing                0.027293
Human Resources           0.026535
Customer Support          0.018196
Name: proportion, dtype: float64

Okay department is fully labeled, but as you can see The Label "Other" makes 47% of all Entries. 


In [26]:
df[df["department"] == "Other"]["position"].value_counts().head(15)


position
Geschäftsführer                      41
Managing Director                    22
CEO                                  20
Owner                                12
Director                             11
Vice President                       11
General Manager                      10
Unternehmensinhaber                   9
Geschäftsführender Gesellschafter     9
Professor                             8
Rechtsanwalt                          8
Intern                                7
CFO                                   7
Research Assistant                    7
Chief Executive Officer               6
Name: count, dtype: int64

In [25]:
df["seniority"].value_counts(normalize=True)

seniority
Professional    0.462092
Lead            0.174375
Management      0.158453
Junior          0.087187
Senior          0.064443
Director        0.053450
Name: proportion, dtype: float64

So as you can see there the seniority is complete. 